### PR TITLE
Support CreateQuery in AMC

### PR DESCRIPTION
**Pull request by pakrym**

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.Query/src/LogsQueryClient.cs [hunk 1]
```diff
@@ -187,6 +191,110 @@ public virtual async Task<Response<LogsBatchQueryResult>> QueryBatchAsync(LogsBa
                 throw;
             }
         }
+
+        /// <summary>
+        /// Create a Kusto query from an interpolated string.  The interpolated values will be quoted and escaped as necessary.
+        /// </summary>
+        /// <param name="filter">An interpolated query string.</param>
+        /// <returns>A valid Kusto query.</returns>
+        public static string CreateQuery(FormattableString filter)
```
Comment (christothes) [inline sdk/monitor/Azure.Monitor.Query/src/LogsQueryClient.cs:200]: Are verbatim strings supported also? ex. `$@"""quoted string {value}"""`

File: sdk/monitor/Azure.Monitor.Query/src/LogsQueryClient.cs [hunk 2]
```diff
@@ -187,6 +191,110 @@ public virtual async Task<Response<LogsBatchQueryResult>> QueryBatchAsync(LogsBa
...
+                {
+                    // Null
+                    null => throw new ArgumentException(
+                        $"Unable to convert argument {i} to an Kusto literal. " +
+                        $"Unable to format an untyped null value, please use typed-null expression " +
+                        $"(bool(null), datetime(null), dynamic(null), guid(null), int(null), long(null), real(null), double(null), time(null))"),
+
+                    // Boolean
+                    true => "true",
+                    false => "false",
```
Comment (christothes) [inline sdk/monitor/Azure.Monitor.Query/src/LogsQueryClient.cs:217]: Should this be `bool(true|false)`?

File: sdk/monitor/Azure.Monitor.Query/tests/LogsQueryClientClientLiveTests.cs [hunk 3]
```diff
@@ -537,6 +537,54 @@ await client.QueryAsync(TestEnvironment.WorkspaceId, $"range x from 1 to {cnt} s
             }
         }
 
+        [RecordedTest]
+        [TestCaseSource(nameof(Queries))]
+        public async Task CanQueryWithFormattedQuery(FormattableString query)
```
Comment (christothes) [inline sdk/monitor/Azure.Monitor.Query/tests/LogsQueryClientClientLiveTests.cs:542]: clever test!

File: sdk/monitor/Azure.Monitor.Query/src/LogsQueryClient.cs [hunk 4]
```diff
@@ -187,6 +191,110 @@ public virtual async Task<Response<LogsBatchQueryResult>> QueryBatchAsync(LogsBa
...
+                        escaped.Append("\\\"");
+                        break;
+                    case '\\':
+                        escaped.Append("\\\\");
+                        break;
+                    case '\r':
+                        escaped.Append("\\r");
+                        break;
+                    case '\n':
+                        escaped.Append("\\n");
```
Comment (christothes) [inline sdk/monitor/Azure.Monitor.Query/src/LogsQueryClient.cs:276]: Does this work with [multi-line string literals](https://docs.microsoft.com/en-us/azure/data-explorer/kusto/query/scalar-data-types/string#multi-line-string-literals)?

Comment (pakrym) [reply sdk/monitor/Azure.Monitor.Query/src/LogsQueryClient.cs:276]: We don't generate multi-line literals, only the normal ones `"..."` with newlines escaped.

File: sdk/monitor/Azure.Monitor.Query/src/LogsQueryClient.cs (hunk 2 above)
Comment (pakrym) [reply sdk/monitor/Azure.Monitor.Query/src/LogsQueryClient.cs:217]: True and false carry the type information on their own. having the wrapping type is more important where there is parsing or ambiguous type involved.

File: sdk/monitor/Azure.Monitor.Query/src/LogsQueryClient.cs (hunk 1 above)
Comment (pakrym) [reply sdk/monitor/Azure.Monitor.Query/src/LogsQueryClient.cs:200]: Yes but not your example. In your example, the value is inserted inside the Kusto string literal. The expectation is that every formatting parameter is a distinct value of its own.

**Review by christothes — APPROVED** [review]

**Pull request merged**